python/cuml/internals/api_context_managers.py

```diff
@@ -153,33 +208,58 @@ def output_type(self, value: str):
 
     @property
     def memory_type(self):
+        """
+        MemoryType: The current memory type for the internal API context.
+        """
         return self._memory_type
 
     @memory_type.setter
     def memory_type(self, value):
         self._memory_type = MemoryType.from_str(value)
 
     def pop_all(self):
-        """Preserve the context stack by transferring it to a new instance."""
+        """
+        Preserve the context stack by transferring it to a new instance.
+
+        Returns
+        -------
+        contextlib.ExitStack
+            A new context stack with the same exit callbacks.
+        """
         new_stack = contextlib.ExitStack()
         new_stack._exit_callbacks = self._exit_callbacks
         self._exit_callbacks = deque()
         return new_stack
 
     def __enter__(self) -> int:
-
+        """
+        Enter the internal API context.
```

Comment:
Explain what this actually does. It should be clear that we are entering a context within this call.